### PR TITLE
AP_HAL_ChibiOS: replace NO_FASTBOOT with AP_FASTBOOT_ENABLED (NFC)

### PR DESCRIPTION
**Pull request by peterbarker**
This is just a rename for `NO_FASTBOOT` to `AP_FASTBOOT_ENABLED`.

Looking at this, it's unclear to me how the iofirmware gets `has_watchdog_safety_off` correct.  It relies on `persistent_data.safety_state` - but we load/save this state using `get_rtc_backup` - and that functionality is gated on `NO_FASTBOOT` in master (which iomcu turns off).

```
Board              AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                      *      *           *       *                 *      *      *
Hitec-Airspeed     0                                                                     
KakuteH7-bdshot               *      *           *       *                 *      *      *
MatekF405                     *      *           *       *                 *      *      *
Pixhawk1-1M                   *      *           *       *                 *      *      *
f103-QiotekPeriph  0                                                                     
f303-Universal     0                                                                     
iomcu                                                          *                         
revo-mini                     *      *           *       *                 *      *      *
skyviper-v2450                                   *                                       
```

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/iomcu_f103_8MHz/hwdef.dat [hunk 1]
```diff
@@ -132,7 +132,7 @@ define DMA_RESERVE_SIZE 0
 define SOFTSIG_MAX_SIGNAL_TRANSITIONS 128
 
 define IOMCU_FW TRUE
-define NO_FASTBOOT
+define AP_FASTBOOT_ENABLED
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/iomcu_f103_8MHz/hwdef.dat:135]: should be 0?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/iomcu_f103_8MHz/hwdef.dat (hunk 1 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/iomcu_f103_8MHz/hwdef.dat:135]: worth working out why CI didn't catch this as a build error

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/iomcu_f103_8MHz/hwdef.dat:135]: we need to use Tools/scripts/build_iofirmware.py in CI

Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/iomcu_f103_8MHz/hwdef.dat]: > should be 0?

Fixed.

Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/iomcu_f103_8MHz/hwdef.dat]: > we need to use Tools/scripts/build_iofirmware.py in CI

https://github.com/ArduPilot/ardupilot/pull/23060

**Comment by peterbarker** [comment]
Fixed the missing boolean 1 in the define, and ran the generate-iomcu-firmware script locally to verify they worked.  I've also merged the PR which moves us to using that generate script as part of CI - this is rebased on top of that.

**Review by tridge — APPROVED** [review]

**Pull request merged**